src/Pate/Verification/SymbolicExecution.hs

```diff
@@ -63,17 +64,14 @@ archStructRepr = do
 -- Note that we consider any error blocks as terminal. We also consider any arch
 -- term statement as terminal; those are usually some kind of
 -- architecture-specific trap instruction.
-isTerminalBlock :: MD.ParsedBlock arch ids -> Bool
-isTerminalBlock pb = case MD.pblockTermStmt pb of
-  MD.ParsedCall{} -> True
-  MD.PLTStub{} -> True
-  MD.ParsedJump{} -> False
-  MD.ParsedBranch{} -> False
-  MD.ParsedLookupTable{} -> False
-  MD.ParsedReturn{} -> False
-  MD.ParsedArchTermStmt{} -> True -- TODO: think harder about this
-  MD.ParsedTranslateError{} -> True
-  MD.ClassifyFailure{} -> True
+isTerminalBlock :: MCS.HasArchEndCase arch => MD.ParsedBlock arch ids -> Bool
+isTerminalBlock pb =
+  let MCS.MacawBlockEnd c _ = MCS.termStmtToBlockEnd (MD.pblockTermStmt pb)
+  in case c of
+    MCS.MacawBlockEndCall -> True
+    MCS.MacawBlockEndArch -> True
+    MCS.MacawBlockEndFail -> True
+    _ -> False
```

Comment:
Can you make this into a complete match (i.e., matching on all of the constructors)? It is really useful if the compiler can yell at us if we add a new one and make us think about what the behavior should be